### PR TITLE
Save torrent state only if state has changed. 

### PR DESCRIPTION
**Pull request by ratoaq2**
Running deluge in a PC or NAS server prevents HDD to spin down, even if there is no activity, no torrents or whatsoever. This happens because deluge writes the torrent state to a file every time.

This PR will make sure that this file is only written if the torrent state has actually changed.

### CODE REVIEW

File: deluge/core/torrentmanager.py [hunk 1]
```diff
@@ -157,6 +173,9 @@ def __init__(self):
         # Workaround to determine if TorrentAddedEvent is from state file
         self.session_started = False
 
+        # Keep the last saved state
+        self.last_saved_state = None
```
Comment (cas--) [inline deluge/core/torrentmanager.py]: I'd prefer it called `prev_saved_state` as a loose consistency ;)

File: deluge/core/torrentmanager.py [hunk 2]
```diff
@@ -111,10 +111,26 @@ def __init__(self,
         self.move_completed = move_completed
         self.move_completed_path = move_completed_path
 
+    def __eq__(self, other):
+        if not isinstance(other, TorrentState):
```
Comment (cas--) [inline deluge/core/torrentmanager.py]: Would prefer a `try..except` here. It is more likely to be an instance of TorrentState than not, and bit more pythonic.

File: deluge/core/torrentmanager.py [hunk 3]
```diff
@@ -111,10 +111,26 @@ def __init__(self,
...
+
+    def __ne__(self, other):
+        return not self == other
+
 class TorrentManagerState:
     def __init__(self):
         self.torrents = []
 
+    def __eq__(self, other):
+        if not isinstance(other, TorrentManagerState):
```
Comment (cas--) [inline deluge/core/torrentmanager.py]: Same use `try..except` as above

File: deluge/core/torrentmanager.py (hunk 2 above)
Comment (ratoaq2) [inline deluge/core/torrentmanager.py]: I didn't fully get you here:

```
    def __eq__(self, other):
        try:
            return self.__dict__ == other.__dict__
        except:
            return False
```

In theory the following could still happen:

```
class A:
    def __init__(self):
        a = 'a'
    
class B:
    def __init__(self):
        a = 'a'

A().__dict__ == B().__dict__
True
```

Comment (cas--) [inline deluge/core/torrentmanager.py]: Fair point. I do prefer a positive isinstance statement and you could do it in one line:

```python    
def __eq__(self, other):
    return isinstance(other, TorrentState) and self.__dict__ == other.__dict__
```

**Comment by ratoaq2** [comment]
Done

**Comment by cas--** [comment]
That's great thanks. One final thing, could you update the commit message with [Core] and move the second sentence out of the summary.

http://dev.deluge-torrent.org/wiki/GitTips#CommitMessages

**Comment by ratoaq2** [comment]
Done

**Pull request merged**